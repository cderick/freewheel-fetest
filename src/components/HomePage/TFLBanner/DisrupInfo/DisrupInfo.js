import React from 'react'
import PropTypes from 'prop-types'
import s from './DisrupInfo.module.scss'

class DisrupInfo extends React.Component {
    render() {
        if (!this.props.transportSelect) return <div className={`col-5 m-5 ${s.infoService}`}></div>
        const { transportSelect } = this.props
        return (
            <div className={`col-5 m-5 ${s.infoService}`}>
                {transportSelect &&
                    <section>
                        <p className={s.linesClass}><strong className="h5">Line:</strong> {transportSelect.name}</p>
                        {(transportSelect.lineStatuses.filter(status => status.statusSeverity === 10)).length > 0
                            ? (
                                <p className={s.allClear}>No service disruptions</p>
                            ) : (
                                <div>
                                    <p className={s.notWorking}>Service currently suffering disruptions</p>
                                    {transportSelect.lineStatuses.map((disrupt, disIn) => (
                                        <section key={`dis${disIn}`}>
                                            <p><strong className="h5">Reason: </strong>{disrupt.reason}</p>
                                        </section>
                                    ))}
                                </div>
                            )}
                    </section>
                }
            </div>

        )
    }
}

DisrupInfo.propTypes = {
    transportSelect: PropTypes.objectOf(PropTypes.any),
}

DisrupInfo.defaultProps = {
    transportSelect: null,
}

export default DisrupInfo
