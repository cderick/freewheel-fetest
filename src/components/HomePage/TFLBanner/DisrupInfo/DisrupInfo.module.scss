.infoService {
  background-color: #eaeaea;

  .linesClass {
    padding-top: 15px;
  }

  .allClear {
    background-color: #caffed;
    padding: 20px 10px;
  }

  .notWorking {
    padding: 20px 10px;
    background-color: #ff8a8a;
  }
}
