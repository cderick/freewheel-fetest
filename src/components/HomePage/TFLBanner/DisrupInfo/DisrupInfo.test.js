import React from 'react'
import renderer from 'react-test-renderer'
import DisrupInfo from './DisrupInfo'
import mockData from './__mock__/mock'

const componentNoprops = renderer.create(<DisrupInfo />)
const componentWithProps = renderer.create(<DisrupInfo {...mockData}  />)

describe('Test on <DisrupInfo /> component', () => {
   test('should not render with no props', () => {
      const treeNoprops = componentNoprops.toJSON()
      expect(treeNoprops).toMatchSnapshot()
   })
   test('should render with all props', () => {
      const treeWithprops = componentWithProps.toJSON()
      expect(treeWithprops).toMatchSnapshot()
   })
})