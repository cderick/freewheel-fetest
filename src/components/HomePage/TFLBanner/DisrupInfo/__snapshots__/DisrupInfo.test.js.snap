// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on <DisrupInfo /> component should not render with no props 1`] = `
<div
  className="col-5 m-5 infoService"
/>
`;

exports[`Test on <DisrupInfo /> component should render with all props 1`] = `
<div
  className="col-5 m-5 infoService"
>
  <section>
    <p
      className="linesClass"
    >
      <strong
        className="h5"
      >
        Line:
      </strong>
       
      Bakerloo
    </p>
    <p
      className="allClear"
    >
      No service disruptions
    </p>
  </section>
</div>
`;
